### PR TITLE
Accounts bloom filter

### PR DESCRIPTION
**Pull request by NikVolf**
on top of @arkpar #2308 

- performace untested

still to do:
- add migration, easy (right now using this requires *re-sync*)

### CODE REVIEW

**Comment by arkpar** [comment]
128k is not enough. for 1m accounts a bloom of 10mbit is required, here's the calculator: http://hur.st/bloomfilter?n=1000000&p=1.0E-2
Ideally it should be dynamically sized depending on the number of items.

**Comment by NikVolf** [comment]
In this algorithm we use increase in bloom space also leads to increase in number of hash functions
for master probably need completely different bloom, yes
I will try to calibrate with large bloom

**Comment by arkpar** [comment]
Number of hash functions only affects error rate. It does not depend on the number of items

**Comment by arkpar** [comment]
Instead of checking for bloom in every method of the state it can be done in `ensure_cache` and `require_or_from`

**Comment by arkpar** [comment]
`note_account_bloom` should only be called in `commit` for dirty accounts

File: ethcore/src/state_db.rs [hunk 1]
```diff
@@ -16,38 +16,92 @@
...
 		}
 	}
 
 	/// Commit all recent insert operations and canonical historical commits' removals from the
 	/// old era to the backing database, reverting any non-canonical historical commit's inserts.
 	pub fn commit(&mut self, batch: &DBTransaction, now: u64, id: &H256, end: Option<(u64, H256)>) -> Result<u32, UtilError> {
-		self.db.commit(batch, now, id, end)
+		// commit bloom
+		let transaction = DBTransaction::new(self.db.backing());
+		try!(transaction.put(None, b"accounts_bloom", &self.account_bloom.lock()));
```
Comment (arkpar) [inline ethcore/src/state_db.rs]: Should be part of the same transaction as `commit`

**Pull request closed without merging**